src/impl/realm_coordinator.cpp

```diff
@@ -148,35 +147,52 @@ void RealmCoordinator::set_config(const Realm::Config& config)
 
 std::shared_ptr<Realm> RealmCoordinator::get_realm(Realm::Config config)
 {
-    std::lock_guard<std::mutex> lock(m_realm_mutex);
+    std::unique_lock<std::mutex> lock(m_realm_mutex);
 
     set_config(config);
 
+    auto schema = std::move(config.schema);
+    auto migration_function = std::move(config.migration_function);
+    config.schema = {};
+
+    std::shared_ptr<Realm> realm;
     if (config.cache) {
         AnyExecutionContextID execution_context(config.execution_context);
         for (auto& cached_realm : m_weak_realm_notifiers) {
-            if (cached_realm.is_cached_for_execution_context(execution_context)) {
-                // can be null if we jumped in between ref count hitting zero and
-                // unregister_realm() getting the lock
-                if (auto realm = cached_realm.realm()) {
-                    return realm;
-                }
+            if (!cached_realm.is_cached_for_execution_context(execution_context))
+                continue;
+            realm = cached_realm.realm();
+            // can be null if we jumped in between ref count hitting zero and
+            // unregister_realm() getting the lock
+            if (realm)
+                break;
+        }
+    }
+    if (!realm) {
+        realm = Realm::make_shared_realm(std::move(config), shared_from_this());
+        if (!config.read_only() && !m_notifier && config.automatic_change_notifications) {
+            try {
+                m_notifier = std::make_unique<ExternalCommitHelper>(*this);
+            }
+            catch (std::system_error const& ex) {
+                lock.unlock();
```

Comment:
I assume this is necessary to ensure that the lock is dropped prior to `realm` being destroyed as part of stack unwinding?